code/missionui/redalert.cpp

```diff
@@ -885,8 +885,14 @@ void red_alert_bash_wingman_status()
 							mprintf(("Invalid ship class specified in red alert data for ship %s. Using mission defaults.\n", shipp->ship_name));
 					}
 
+					float max_hull;
+					if (shipp->special_hitpoints)
+						max_hull = shipp->ship_max_hull_strength;
+					else
+						max_hull = Ship_info[shipp->ship_info_index].max_hull_strength;
+
```

Comment:
Would an `Assertion()` that `max_hull` is something sane be useful, say `> 0.0f`? Ditto for similar-looking code below.

Reply:
I don't think this is necessary.  It already happens before this in the code flow.